newrelic-metrics/firehose.tf

```diff
@@ -0,0 +1,136 @@
+# https://docs.newrelic.com/docs/infrastructure/amazon-integrations/connect/aws-metric-stream-setup/
+data "aws_caller_identity" "default" {
+}
+
+
+module "firehose_bucket" {
+  source = "../private-bucket"
+  bucket_name = "${var.name_prefix}-newrelic-firehose-data"
+  tags = var.tags
+}
```

Comment:
I'm suddenly realizing that the last time I tried to use `private-bucket`, the plan looked good and then the apply step broke because we don't set the appropriate object ownership configuration to enable ACLs. I fear that we're gonna publish this module and then it's gonna bomb out trying to create the resources.
I might try updating the `private-bucket` module to the following:

```diff
diff --git a/private-bucket/main.tf b/private-bucket/main.tf
index e293e25..b0c36e2 100644
--- a/private-bucket/main.tf
+++ b/private-bucket/main.tf
@@ -11,9 +11,19 @@ resource "aws_s3_bucket" "default" {
   )
 }
 
+resource "aws_s3_bucket_ownership_controls" "default" {
+  bucket = aws_s3_bucket.default.id
+
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
 resource "aws_s3_bucket_acl" "default" {
   bucket = aws_s3_bucket.default.id
   acl = "private"
+
+  depends_on = [aws_s3_bucket_ownership_controls.default]
 }
 
 resource "aws_s3_bucket_server_side_encryption_configuration" "default" {
~
```

Reply:
This is based off of these docs: 
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html#object-ownership-overview
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_acl#with-private-acl